src/lambdaisland/embedkit.clj

```diff
@@ -69,7 +69,9 @@
 
 (defprotocol IConnection
   (path-url [_ p])
-  (request [_ method url opts]))
+  (request [_ method url opts])
+  (conn-cache [_])
+  (secret-key [_]))
```

Comment:
There were two things for which we accessed the connection object directly: `:cache`, and `:secret-key`. By adding these to the protocol instead `conn` becomes a well defined abstraction, that is accessed only through its interface methods, so we can have polymorphism.